bpf/lib/nodeport.h

```diff
@@ -763,6 +763,13 @@ static __always_inline bool nodeport_nat_ipv4_needed(struct __ctx_buff *ctx,
 			struct remote_endpoint_info *info;
 			*from_endpoint = true;
 
+#ifdef IPV4_NATIVE_ROUTING_CIDR
+			if (ipv4_is_in_subnet(ip4->daddr,
+					      IPV4_NATIVE_ROUTING_CIDR,
+					      IPV4_NATIVE_ROUTING_CIDR_LEN))
+				return false;
+#endif
```

Comment:
Can't we save the cycles of the map lookup and do this before `__lookup_ip4_endpoint()`?

Reply:
`__lookup_ip4_endpoint()` is for detecting whether src is a local pod, while the added native-routing-cidr check is for detecting whether dst is a pod in a cluster (local or remote).